Navigation/include/ActsLUXEPipeline/LUXEMeasurement.hpp

```diff
@@ -0,0 +1,135 @@
+#include <fstream>
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/archive/binary_iarchive.hpp>
+
+/// Propagator action to create measurements.
+namespace LUXENavigator {
+
+// Store only essential components
+    struct MeasurementComponents {
+        int eId;
+        int gId;
+        double l0;
+        double l1;
+        double phi;
+        double theta;
+        double qOverP;
+        double time;
+        double trueVertexX;
+        double trueVertexY;
+        double trueVertexZ;
+        double trueVertexT;
+
+        MeasurementComponents() :
+                eId(),
+                gId(),
+                l0(),
+                l1(),
+                phi(),
+                theta(),
+                qOverP(),
+                time(),
+                trueVertexX(),
+                trueVertexY(),
+                trueVertexZ(),
+                trueVertexT() {}
+
+        MeasurementComponents(LUXEDataContainer::SimMeasurement m) :
+                eId(m.sourceLink.get<SimpleSourceLink>().eventId),
+                gId(m.sourceLink.get<SimpleSourceLink>().geometryId().sensitive()),
+                l0(m.sourceLink.get<SimpleSourceLink>().parameters[0]),
+                l1(m.sourceLink.get<SimpleSourceLink>().parameters[1]),
+                phi(m.truthParameters[Acts::eBoundPhi]),
+                theta(m.truthParameters[Acts::eBoundTheta]),
+                qOverP(m.truthParameters[Acts::eBoundQOverP]),
+                time(m.truthParameters[Acts::eBoundTime]),
+                trueVertexX(m.trueVertex[0]),
+                trueVertexY(m.trueVertex[1]),
+                trueVertexZ(m.trueVertex[2]),
+                trueVertexT(m.trueVertex[3]){}
+
+
+        template<class Archive>
+        void serialize(Archive& ar, const unsigned int version) {
+            ar & eId;
+            ar & gId;
+            ar & l0;
+            ar & l1;
+            ar & phi;
+            ar & theta;
+            ar & qOverP;
+            ar & time;
+            ar & trueVertexX;
+            ar & trueVertexY;
+            ar & trueVertexZ;
+            ar & trueVertexT;
+        }
+    };
+
+    void saveMeasurementsToFile(const LUXEDataContainer::SimMeasurements& measurements, const std::string& filename) {
+        std::ofstream ofs(filename, std::ios::binary);
+        if (!ofs) {
+            std::cerr << "Error opening file for writing: " << filename << std::endl;
+            return;
+        }
+        boost::archive::binary_oarchive oa(ofs);
+        size_t size=measurements.size();
+        oa << size; // Write the size of the vector
+
+        for (const auto& measurement : measurements) {
+            MeasurementComponents m_comp(measurement);
+            oa << m_comp; // Serialize each Measurements object
+        }
+        ofs.flush();
+    }
+
+// Function to load vector of Measurements from a file
+    LUXEDataContainer::SimMeasurements loadMeasurementsFromFile(const std::string& filename) {
```

Comment:
We already have a TTree reader. I don't mind supporting another format, but I feel like we have to have an implementation that is `ROOTFileReader` compatible